src/main/java/com/smartcar/sdk/SmartcarException.java

```diff
@@ -84,7 +85,16 @@ public String getRequestId() {
   }
 
   public static SmartcarException Factory(final Response response) throws IOException {
-    JsonObject body = gson.fromJson(response.body().string(), JsonObject.class);
+    JsonObject body;
+    try {
+      body = gson.fromJson(response.body().string(), JsonObject.class);
+    }catch (com.google.gson.JsonParseException exception) {
+      // In case the body is a string. Ex. gateway timeout where LB sends a non json body
+      String errorBody = Json.createObjectBuilder()
+              .add("message", response.body().string())
+              .build().toString();
+      body = gson.fromJson(errorBody, JsonObject.class);
+    }
```

Comment:
add another generic `catch` for any other error that could be thrown and raise it as a `SmartcarException`